src/hotspot/share/memory/universe.cpp

```diff
@@ -318,6 +318,12 @@ void Universe::genesis(TRAPS) {
   ResourceMark rm(THREAD);
   HandleMark   hm(THREAD);
 
+  // Explicit null checks are needed if these offsets are larger than the page size
```

Comment:
Nit: the condition checks smaller, but the comment says larger, which leaves "equal to the page size" unclear leading the reader to wonder "Is the comment wrong or the condition check?".

Suggestion: s/larger/not smaller/